applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,170 @@
+            "second_alpha_value"         : 1.0,
+            "min_alpha"                  : 0.1,
+            "max_alpha"                  : 2.0,
+            "line_search_tolerance"      : 0.5,
+            "rotation_dofs"              : false,
+            "block_builder"              : true,
+            "search_neighbours_step"     : false,
+            "linear_solver_settings":{
+                "solver_type": "AMGCL",
+                "tolerance": 1.0e-6,
+                "max_iteration": 100,
+                "scaling": false,
+                "verbosity": 0,
+                "preconditioner_type": "ILU0Preconditioner",
+                "smoother_type": "ilu0",
+                "krylov_type": "gmres",
+                "coarsening_type": "aggregation"
+            },
+            "problem_domain_sub_model_part_list": [""],
+            "processes_sub_model_part_list": [""],
+            "body_domain_sub_model_part_list": [""],
+            "loads_sub_model_part_list": [],
+            "loads_variable_list": []
+        }""")
+
+        this_defaults.AddMissingParameters(super().GetDefaultParameters())
+        return this_defaults
+
+    def PrepareModelPart(self):
+        super().PrepareModelPart()
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Model reading finished.")
+
+    def AddDofs(self):
+        KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.TEMPERATURE, self.main_model_part)
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "DOFs added correctly.")
+
+    def Initialize(self):
+        KratosMultiphysics.Logger.PrintInfo("::[GeoMechanics_T_Solver]:: ", "Started Initialization")
+        
+        super().Initialize()
+
+        # Check if everything is assigned correctly
+        self.Check()
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Solver initialization finished and validated.")
+
+    def _ConstructScheme(self, scheme_type, solution_type):
+
+        self.main_model_part.ProcessInfo.SetValue(KratosGeo.DT_TEMPERATURE_COEFFICIENT, 1.0)
+
+        if scheme_type.lower() == "newmark" or scheme_type.lower() == "newmark_flow":
+            theta = self.settings["newmark_theta"].GetDouble()
+            KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, solution_type", solution_type)
+            if solution_type.lower() == "transient-heat-transfer" or solution_type.lower() == "transient_heat_transfer":
+                KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, scheme", "Newmark Transient heat transfer.")
+                scheme = KratosGeo.NewmarkTScheme(theta)
+            elif solution_type.lower() == "steady-state-heat-transfer" or solution_type.lower() == "steady_state_heat_transfer":
+                raise Exception("Steady state heat transfer calculations are not supported.")
+            else:
+                raise Exception("Undefined solution type", solution_type)
```

Comment:
I would suggest to raise a `RuntimeError` rather than this base exception class (I suspect that SonarQube will also complain about the use of `Exception`).